[BUG] Build failing for Spark 3.3.0 due to deprecated method warnings

Build is failing with this error:
```
[2022-01-26T17:01:13.488Z] [INFO] Compiling 247 Scala sources and 28 Java sources to /home/jenkins/agent/workspace/jenkins-rapids_nightly-pre_release-github-335/sql-plugin/target/spark330/classes ...
[2022-01-26T17:01:35.380Z] [ERROR] [Error] /home/jenkins/agent/workspace/jenkins-rapids_nightly-pre_release-github-335/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala:128: method newTaskTempFile in class FileCommitProtocol is deprecated (since 3.3.0): use newTaskTempFile(..., spec: FileNameSpec) instead
[2022-01-26T17:01:35.380Z] [ERROR] [Error] /home/jenkins/agent/workspace/jenkins-rapids_nightly-pre_release-github-335/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala:262: method newTaskTempFileAbsPath in class FileCommitProtocol is deprecated (since 3.3.0): use newTaskTempFileAbsPath(..., spec: FileNameSpec) instead
[2022-01-26T17:01:35.380Z] [ERROR] [Error] /home/jenkins/agent/workspace/jenkins-rapids_nightly-pre_release-github-335/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala:264: method newTaskTempFile in class FileCommitProtocol is deprecated (since 3.3.0): use newTaskTempFile(..., spec: FileNameSpec) instead
[2022-01-26T17:01:35.380Z] [ERROR] three errors found
```

This was triggered by a recent change in Spark 3.3, see [SPARK-38015](https://issues.apache.org/jira/browse/SPARK-38015)

--- a/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala
+++ b/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala
@@ -120,6 +120,9 @@ class GpuSingleDirectoryDataWriter(
   // Initialize currentWriter and statsTrackers
   newOutputWriter()
 
+  @scala.annotation.nowarn(
+    "msg=method newTaskTempFile in class FileCommitProtocol is deprecated"
+  )
   private def newOutputWriter(): Unit = {
     recordsInFile = 0
     releaseResources()
@@ -245,6 +248,9 @@ class GpuDynamicPartitionDataWriter(
     row => proj(row).getString(0)
   }
 
+  @scala.annotation.nowarn(
+    "msg=method newTaskTempFile.* in class FileCommitProtocol is deprecated"
+  )
   private def newOutputWriter(partDir: String): Unit = {
     recordsInFile = 0
     releaseResources()

